pgcommitfest/commitfest/models.py

```diff
@@ -102,11 +102,37 @@ def __str__(self):
         return self.version
 
 
+class ColorField(models.CharField):
+    """
+    A small wrapper around a CharField that can hold a #RRGGBB color code. The
+    primary reason to have this wrapper class is so that the TagAdmin class can
+    explicitly key off of it to inject a color picker in the admin interface.
+    """
+
+    def __init__(self, *args, **kwargs):
+        kwargs["max_length"] = 7  # for `#RRGGBB` format
+        super().__init__(*args, **kwargs)
+
+
+class Tag(models.Model):
+    """Represents a tag/label on a patch."""
+
+    name = models.CharField(max_length=50, unique=True)
+    color = ColorField()
+
+    class Meta:
+        ordering = ("name",)
+
+    def __str__(self):
+        return self.name
+
+
 class Patch(models.Model, DiffableModel):
     name = models.CharField(
         max_length=500, blank=False, null=False, verbose_name="Description"
     )
     topic = models.ForeignKey(Topic, blank=False, null=False, on_delete=models.CASCADE)
+    tags = models.ManyToManyField(Tag, related_name="patches", blank=True)
```

Comment:
Maybe also pre-mature, and likely reasonably added later, but I envisioned an ability for annotations to be added to the tag-patch mapping.  Maybe the commit message and email threads are enough places to write stuff but, especially for something like 'Good First Issue', some commentary as to why would be expected to exist somewhere, and a tag annotation seems like a good place.

Reply:
I'm also thinking we'll use the mailing list for primary discussion while this shakes out, but I think this would be a good thing to add eventually.